### PR TITLE
Fixing CI Issues

### PR DESCRIPTION
**Pull request by greglucas**
edit: I went to fix one thing and then more things broke in CI.

Commit 1: owslib pin to 0.19.2 for Python 3.5
Commit 2: Updating docs for successful gallery build because of the fix in owslib 0.19.2
Commit 3: Fixing Stickler imports not at top of file.
Commit 4: Updating conda in Appveyor Python 3.6 build and pinning Cython for Python 2 on Travis.

----

owslib 0.19.2 dropped Python 3.5 support.
https://geopython.github.io/OWSLib/#id5
This pins owslib to 0.19.1 for the Python 3.5 build on CI.

## Implications
I'm not sure if there is a better if statement to put this on for CI builds. This will put it on the install line twice for the Python 3.5 PACKAGES under the conda install (conda install owslib=0.19.1 owslib), but conda should be able to figure the pin out.

### CODE REVIEW

**Comment by greglucas** [comment]
and wouldn't you guess, Python 3.6 and Appveryor are now failing :(

I restarted one and it fixed itself. Recycling here to try again.

**Comment by greglucas** [comment]
Appveyor and docs are now unhappy. If anyone else wants to take a stab at this, feel free to push to my branch or just pin owslib=0.19.1 to fix Python 3.5 and expand on your own branch.

## Docs
The build looks like it could be due to using an old sphinx-gallery (may need to go 0.3 or newer).

```
build succeeded, 4 warnings.

The HTML pages are in build/html.

Exception occurred:
  File "/opt/conda/envs/test-environment/lib/python3.7/site-packages/sphinx_gallery/gen_gallery.py", line 313, in sumarize_failing_examples
    "\n" + "-" * 79)
ValueError: Here is a summary of the problems encountered when running the examples

Examples expected to fail, but not failing:
Please remove these examples from
sphinx_gallery_conf['expected_failing_examples']
/root/project/docs/source/cartopy/examples/reprojected_wmts.pyin your conf.py file
/root/project/docs/source/cartopy/examples/wmts_time.pyin your conf.py file
/root/project/docs/source/cartopy/examples/wmts.py
```

## Appveyor
Something with conda packages possibly?
```
ERROR conda.core.link:_execute(502): An error occurred while installing package 'conda-forge::atomicwrites-1.3.0-py_0'.
```

**Comment by QuLogic** [comment]
Python 2 docs would be restricted to owslib 0.18, while Python 3 has the latest, with fixes for #1384.

File: docs/source/conf.py [hunk 1]
```diff
@@ -97,6 +97,19 @@
 # The full version, including alpha/beta/rc tags.
 release = cartopy.__version__
 
+import owslib
```
[the remark this answers is not in the document]
Comment (QuLogic) [reply docs/source/conf.py]: Going to ignore these, because it's already all over the place in this file.

**Comment by greglucas** [comment]
Thanks for that! I should have read the error messages a little closer :)

I updated the module imports to fix Stickler. Now all that is left is fixing Appveyor... I have no expertise on the windows side of things, so help is welcome from anyone on that.

Only Python3.6 is failing and it is failing early on in the conda build. Appveyor is also trying to install proj7.0.0, which may cause issues.

**Comment by dopplershift** [comment]
Is Python 3.5 something we really need to be supporting right now? 3.6.0 was released Dec 2016. Much of the scientific Python community has moved to only support minor versions from the [last 42 months](https://numpy.org/neps/nep-0029-deprecation_policy.html). I should also add that conda-forge isn't even building 3.5 packages any more.

**Comment by QuLogic** [comment]
Well, the problem is that this is now building 3.6, but it's failing the same way.

**Comment by dopplershift** [comment]
Well, the title of the PR implies fixing Python 3.5--which I'm advocating to get rid of entirely.

Current build failure on AppVeyor for 3.6 is just...weird.

**Comment by greglucas** [comment]
@dopplershift, the title was misleading because I thought that was the only fix needed! Then more have just kept popping up. I've updated it now to just fixing CI.

I think an even more glaring problem than Python 3.5 is supporting an EOL Python 2.7 in the CI systems still! Those packages aren't being updated either.

To be clear: I think we should push out a 0.18 release as soon as possible when these CIs pass and then start testing on Py3.6-3.8 only.

**Comment by greglucas** [comment]
There was an issue with older conda versions, so updating conda on Appveyor solved that issue. For some reason, conda also wanted to grab an old version of Cython on the Python 2.7 build within Travis, so I decided to just pin that.

CI seems to be good now!

**Pull request merged**